TW-583 Postman CLI 1.0 (#4245)
postmanlabs/postman-docs: src/pages/docs/postman-cli/postman-cli-options.md

---
title: "Postman CLI command options"
updated: 2022-09-15
contextual_links:
  - type: section
    name: "Prerequisites"
  - type: link
    name: "Postman CLI overview"
    url: "/docs/postman-cli/postman-cli-overview/"
  - type: link
    name: "Installing the Postman CLI"
    url: "/docs/postman-cli/postman-cli-installation/"
  - type: section
    name: "Next steps"
  - type: link
    name: "Running a collection with the Postman CLI"
    url: "/docs/postman-cli/postman-cli-run-collection/"
warning: false
tags:
  - "Postman CLI"

---

Commands and options for using the Postman CLI.

## Contents

* [Commands and options - quick reference](#commands-and-options---quick-reference)
* [Downloading and installing](#downloading-and-installing)
* [Logging and and logging out](#logging-in-and-logging-out)
* [Running collections](#running-collections)

> * [Governance and security](#governance-and-security)

Reviewer: These are enterprise-only options - needs a note

| [**`postman`**](#postman) | The base command. |
| `--help` | Return information about Postman CLI commands and options. |
| `--version` | Return the version number for the Postman CLI.  |

## Downloading and installing

You can download the Postman CLI installation script with the following commands:

### Linux

``` bash
curl -o- "https://dl-cli.pstmn.io/install/linux64.sh" | bash
```

### Mac (Intel chip)

``` bash
curl -o- "https://dl-cli.pstmn.io/install/osx_64.sh" | bash
```

<!--
### Mac (Apple chip)

``` bash
curl -o- "https://dl-cli.pstmn.io/install/osx_arm64.sh" | bash
```
-->

### Windows

``` bash
powershell.exe -NoProfile -InputFormat None -ExecutionPolicy AllSigned -Command "[System.Net.ServicePointManager]::SecurityProtocol = 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://dl-cli.pstmn.io/install/win64.ps1'))" && SET "PATH=%PATH%;C:\Postman CLI\"
```

---

## Logging in and logging out

You can use the Postman CLI to log in and out of Postman with the `login` and `logout` commands, described below:

> ### login

Reviewer: postman login, postman logout

> This command authenticates the user and stores the [Postman API key](/docs/developer/intro-api#generating-a-postman-api-key) in your filesystem. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Reviewer: link to API key, explain

> This command authenticates the user and stores the [Postman API key](/docs/developer/intro-api#generating-a-postman-api-key) in your filesystem. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Reviewer: Remove "stores in your filesystem" and say "caches locally"

> This command authenticates the user and stores the [Postman API key](/docs/developer/intro-api#generating-a-postman-api-key) in your filesystem. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Reviewer 2: Added mention about key expiration. Looking into the second comment.

> This command authenticates the user and stores the [Postman API key](/docs/developer/intro-api#generating-a-postman-api-key) in your filesystem. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Reviewer 2: We already link to the API key in this paragraph.

> This command authenticates the user and stores the [Postman API key](/docs/developer/intro-api#generating-a-postman-api-key) in your filesystem. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Reviewer: ```suggestion
This command authenticates the user and caches the [Postman API key](/docs/developer/intro-api#generating-a-postman-api-key) locally. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.
```

#### Example

``` bash
login --with-api-key ABCD-1234-1234-1234-1234-1234
```

### logout

This command logs you out of Postman and deletes the stored API key.

#### Example

``` bash
logout
```

#### Options

 Option | Details |
|:--|:--|
| `--with-api-key [api-key]` | Authenticate the user with the given API key. |

---

## Running collections

You can run your collections with the `postman collection run` command:

### **postman collection run**

This command runs a collection and sends all run results and responses to Postman servers. You can specify the collection with its file path or Collection ID.

> You can find the collection ID in Postman. First, select **Collections** in the sidebar and select a collection. Then select the information icon <img alt="Information icon" src="https://assets.postman.com/postman-docs/icon-information-v9-5.jpg#icon" width="16px"> in the right context bar to access the collection ID.

#### Examples

``` bash
postman collection run /myCollectionFolderName/myCollectionFile.json

> postman collection run 12345678-12345ab-1234-1ab2-1ab2-ab1234112a12

Reviewer: why is this syntax highlighting?

Reviewer 2: Use `plaintext` as the specified language. This turns off the syntax highlighting.

| `--verbose` | Shows detailed information of collection run and each request sent.|
| `--working-dir [path]`| Sets the path of the working directory to use while reading files with relative paths. Defaults to current directory.

---

## Governance and security

API governance is the practice of applying a defined set of standards consistently across your API design and testing phases of your development process. The Postman CLI includes an API linting command, described below. (This feature is available for [Enterprise teams only](https://www.postman.com/pricing/)).

### postman api lint

This command runs validation checks for governance and security rules against the API definition provided in the Postman config file, a local file, or a UUID. You can only lint single-file definitions. `api lint` shows a warning if unable to find the API ID to send data back to Postman.

> This command supports APIs that are stored on Postman and aren't linked to Git.

#### Example

``` bash
postman api lint my-definition-file.json
postman api lint 8854915-bb7236b2-536e-4bdc-bfa2-fbe2fe1941eb
```

---

## Basic command line options

> ### postman

Reviewer: Move this section first

This is the base command, usually combined with `collection run` or `api lint`.

#### Example

``` bash
postman -v
```

#### Options

| Option | Details |
|:--|:--|

> | --help, -h | Returns information about Postman CLI commands and options.|

Reviewer: these should be in code style

---
title: "Postman CLI command options"
updated: 2022-09-15
contextual_links:
  - type: section
    name: "Prerequisites"
  - type: link
    name: "Postman CLI overview"
    url: "/docs/postman-cli/postman-cli-overview/"
  - type: link
    name: "Installing the Postman CLI"
    url: "/docs/postman-cli/postman-cli-installation/"

warning: false
tags:
  - "Postman CLI"

---

Commands and options for using the Postman CLI.

## Contents

> * [Download](#download)

Reviewer: Looks like these need to be updated to match the headings.

* [Basic](#basic)
* [Login/logout](#login-logout)
* [Run a Postman collection](#run-a-Postman-collection)
* [Lint](#lint)
* [Custom reporters](#custom-reporters)

## Commands

> | Command/Option | Description |

Reviewer 3: @petewitcher 

- login/logout should be in a separate section since they are not command options but commands of their own - they are also introducing a new functionality. We need to also talk about postman api key login there and educate users about how they can generate their api keys. 

- the other command options are all options to `postman collection run` command. These options can't be used without that command. Hence logically, these options should come within the command hierarchy or after it at least.

Author: Thanks for the feedback, I've made this change.

| [**`login`**](#postman-login---with-api-key-api-key) | Log into Postman. |
| [**`logout`**](#postman-logout) | Log out of Postman. |
| [**`postman`**](#postman) | The base command. |
| `--help` | Return information about Postman CLI commands and options. |
| `--version` | Return the version number for the Postman CLI.  |
| [**`postman collection run`**](#postman-collection-run-uuid-or-file-name) | Run a collection with a UUID or a local file. |
| `--bail` | Stop the runner when a test case fails. |
| `--color` | Turn off colored output (auto\|on\|off) (default: "auto") |
| `--cookie-jar` | Specify the file path for a JSON Cookie Jar. Uses `tough-cookie` to deserialize the file. |
| `--delay-request` | Specify a delay (in milliseconds) between requests [number]. |
| `--disable-unicode` | Force the unicode disable option. When supplied, replaces all symbols in the output with their plain text equivalents. |
| `--env-var` | Set environment variables in a key=value format on the command line. |
| `--environment` | Specify a Postman environment as a JSON [file]. |
| `--export-collection` | The path to the file where Postman outputs the final collection file before completing a run. |

> | `--export-cookie-jar` | The path to the file where Newman will output the final cookie jar file before completing a run. Uses `tough-cookie` to serialize the file. |

Author: Fixed

| `--export-environment` | The path to the file where Postman outputs the final environment variables file before completing a run. |
| `--export-globals` | The path to the file where Postman outputs the final global variables file before completing a run. |
| `--folder` | Specify a single folder to run from a collection. |
| `--global-var` | Specify global variables on the command line, in a key=value format. |
| `--globals` | Specify a Postman globals file as JSON [file]. |
| `--ignore-redirects` | Turn off automatic following of `3XX` responses. |
| `--insecure` | Turn off strict SSL. |
| `--iteration-count` | Define the number of iterations to run. |
| `--iteration-data` | Specify a data file to use, either JSON or CSV. |
| `--no-insecure-file-read` | Prevent reading files situated outside the working directory. |
| `--silent` | Turn off terminal output. |

> | `--suppress-exit-code` | Continue running tests even after a failure, but exit with `code=0` |

Reviewer 2: ```suggestion
| `--suppress-exit-code` | Continue running tests even after a failure, but exit with `code=0`. |
```

Author: Fixed

| `--timeout` | Specify the time (in milliseconds) to wait for the entire collection run to complete execution. |
| `--timeout-request` | Specify a request timeout (in milliseconds) for a request. |
| `--timeout-script` | Specify the time (in milliseconds) to wait for scripts to complete execution. |
| `--verbose` | Show the details of the collection run and its data ingestion to Postman. |
| `--working-dir` |Set the path of the working directory to use while reading files with relative paths. |
| `-r` | Run a collection with a custom reporter. |

> | [**`postman api lint`**](#postman-api-lint-uuid-or-file-name)| Run validation checks for governance and security rules against the api definition provided in the Postman config file. |

Reviewer 2: ```suggestion
| [**`postman api lint`**](#postman-api-lint-uuid-or-file-name)| Run validation checks for governance and security rules against the API definition provided in the Postman config file. |
```

| `--disable-postman-reporter` | Don't upload data to Postman after linting. |
| `--only-governance-rules` | Only use governance rules at the time of linting. |
| `--only-security-rules` | Only use security rules at the time of linting. |

## Downloading the installation package

Use the URLs below in your script to download the Postman CLI installation package. You can also download the installation packages manually from the Postman [downloads](https://www.postman.com/downloads/) page.

### Linux

>     https://dl-cli.pstmn.io/download/latest/linux64

Reviewer 2: It looks like the code blocks on this page use spaces instead of back ticks. I believe we should be using back ticks to format code blocks (```). Thanks!

### Mac

    https://dl-cli.pstmn.io/download/latest/osx_64

### Windows

    https://dl-cli.pstmn.io/download/latest/win64

### Example - programmatic download

    curl https://dl-cli.pstmn.io/download/latest/linux/postman-cli.zip -o postman-cli.zip

---

## login --with-api-key <api-key>

Log in with your Postman API key.

### Example

>     login --with-api-key <api-key>

Reviewer 2: Wouldn't the full commands for login and logout begin with `postman`?

### Options

#### --help, -h

Returns information about Postman CLI commands and options.

#### --version, -v

Returns the version number for the Postman CLI.

---

> ## postman login --with-api-key <api-key>

Reviewer: Gatsby is eating everything that's using a greater/less than. (It thinks they are HTML tags).

You can either switch to square brackets (easier) or use `&lt;` and `&gt` (not easier). 

Make sure to search/replace across the whole doc to catch these.

> Log in with your Postman API key.

Reviewer: Is login required? Explain if it's opt-in or opt-out.

Reviewer 4: We should mention that it is required only once in a session (or possibly across sessions 🤔). cc @malvikach

Author: Added "The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.".

> Log in with your Postman API key.

Reviewer: Throw a link here to `https://learning.postman.com/docs/developer/intro-api/#generating-a-postman-api-key`

### Example

    postman -v

### Options

#### --help, -h

Returns information about Postman CLI commands and options.

#### --version, -v

Returns the version number for the Postman CLI.

---

## postman collection run <collection-uid> or <file-path>

> Run a collection with options. Specify the collection with its uid or file path.

Reviewer 2: Other occurrence of "UID" are fully capitalized. Should we make that change here?

Author: Fixed

### Example

    postman collection run --folder /myCollectionFolderName/myCollectionFile.json

### Options

#### --bail

Stop the runner when a test case fails.

#### --color <value>

Turn off colored output (auto\|on\|off) (default: "auto")

#### --cookie-jar [path]

Specify the file path for a JSON Cookie Jar. Uses `tough-cookie` to deserialize the file.

#### --delay-request [number]

Specify a delay (in milliseconds) between requests [number].

#### --disable-unicode

Force the unicode disable option. When supplied, all symbols in the output will be replaced by their plain text equivalents.

#### --environment <uid> or <file-path>

Specify an environment file path or UID.

#### --env-var "[environment-variable-name]=[environment-variable-value]"

Set environment variables in a key=value format on the command line. You can add multiple environment variables using `--env-var` multiple times, for example:

>     `--env-var "key1=value1" --env-var "key2=value2"`.

Reviewer: ```suggestion
    --env-var "key1=value1" --env-var "key2=value2"
```

#### --export-cookie-jar [path]

> The path to the file where Newman will output the final cookie jar file before completing a run. Uses `tough-cookie` to serialize the file.

Reviewer 2: Should "Newman" be changed to "Postman" or "Postman CLI"?

> This command authenticates the user and stores the API key in your filesystem. `login` requires one option, `--with-api-key`, that accepts an API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Reviewer 3: the login command when used with --with-api-key  authenticates the user using the Postman API Key. It also stores the Postman API Key.... [contd rest of the sentence]

important to specify its Postman API Key

Reviewer 3: I think we can also cross-link here to the section on LC abut generating postman api keys.

Author: Changed to: 
This command authenticates the user and stores the Postman API key in your filesystem. `login` requires one option, `--with-api-key`, that accepts the Postman API key. The `login` command is required only once per session. Once you've logged in, you remain logged in until you use the `logout` command.

Author: Link added as well.

```

### logout

This command logs you out of Postman and deletes the stored API key.

#### Example

``` bash
logout
```

#### Options

 Option | Details |
|:--|:--|
| `--with-api-key [api-key]` | Authenticate the user with the given API key. |

---

## Running collections

You can run your collections with the `postman collection run` command:

### **postman collection run**

> This command runs a collection and sends all run results and responses to Postman servers. You can specify the collection with its file path or UID.

Reviewer 3: @petewitcher @MatthewPinsker-postman @jkonrath-postman should we clarify what collection uid is and how users can obtain the same.
This is available in the info section of context bar on the postman app.

#### --ignore-redirects

Turn off automatic following of `3XX` responses.

#### --insecure, -k

Turn off strict SSL.

#### --no-insecure-file-reads

Prevent reading of files situated outside of the working directory.

#### --override-request-order

Used with `-i`. Runs a collection with the specified order of the requests or folders. For example:

    postman collection run <collectionUID> -i <folder1UID> -i <folder2UID> --override-request-order

#### --reporter <myReporter>

Run a collection with a custom reporter.

    Install custom reporters with `npm install <custom-reporter-name>`

#### --reporter-myreporter-<option-name> <option-value>

Run a collection with custom reporter options. Multiple options can be added.

#### --silent

Turn off terminal output.

#### --suppress-exit-code, -x

> Continue running tests even after a failure, but exit with `code=0`

Reviewer: ```suggestion
Continue running tests even after a failure, but exit with `code=0`.
```

#### --timeout [number]

Specify the time (in milliseconds) to wait for the entire collection run to complete execution.

#### --timeout-request [number]

Specify a request timeout (in milliseconds) for a request.

#### --timeout-script [number]

Specify the time (in milliseconds) to wait for scripts to complete execution.

#### --verbose

> Show detailed information of collection run and each request sent.

Reviewer 2: Some of the option descriptions do not match between the table and individual sections towards the bottom. Is this intentional?

Author: Thanks I think I fixed these.

#### --working-dir <path>

Set the path of the working directory to use while reading files with relative paths. Default to current directory.

---

## postman api lint <api-id> or <file-name>

> Run validation checks for governance and security rules against the api definition provided in the Postman config file, a local file, or a UUID. You can only lint single-file definitions. Shows a warning if unable to find `<api-id>` to send data back to Postman.

Reviewer 2: ```suggestion
Run validation checks for governance and security rules against the API definition provided in the Postman config file, a local file, or a UUID. You can only lint single-file definitions. Shows a warning if unable to find `<api-id>` to send data back to Postman.
```

Author: Fixed

> > Supports APIs that are stored on Postman and aren't linked to Git.

Reviewer: ```suggestion
> This command supports APIs that are stored on Postman and aren't linked to Git.
```

> |  `-i [requestUID] or [folderUID]` | Runs only the specified folder UID or request UID from the collection. |

Reviewer 3: this also runs the collection in a specific order as specified by the sequence of -i 's 
@jkonrath-postman @petewitcher @MatthewPinsker-postman

> | `--working-dir [path]`| Sets the path of the working directory to use while reading files with relative paths. Defaults to current directory.

Reviewer 3: Add --silent to this list

---
title: "Postman CLI command options"
updated: 2022-09-15
contextual_links:
  - type: section
    name: "Prerequisites"
  - type: link
    name: "Postman CLI overview"
    url: "/docs/postman-cli/postman-cli-overview/"
  - type: link
    name: "Installing the Postman CLI"
    url: "/docs/postman-cli/postman-cli-installation/"

warning: false
tags:
  - "Postman CLI"

---

Commands and options for using the Postman CLI.

## Contents

* [Commands](#commands)
* [Downloading the installation package](#downloading-the-installation-package)
* [postman](#postman)

> * [postman login](#postman-login---with-api-key-api-key)

Reviewer 2: The relative URL does not seem right. With the dashes in the titled, I'm not exactly sure what the correct URL should be. VS Code should be able to generate this.

> * [postman logout](#postman-logout))

Reviewer 2: ```suggestion
* [postman logout](#postman-logout)
```

> * [postman collection run](#postman-api-lint-api-id-or-file-name(#postman-collection-run-collection-uid-or-file-path)

Reviewer 2: ```suggestion
* [postman collection run or](#postman-collection-run-or)
```

> * [postman api lint](#postman-api-lint-api-id-or-file-name)

Reviewer 2: ```suggestion
* [postman api lint or](#postman-api-lint-or)
```

Author: Fixed

## Commands

> | Command/Option | Description |

Reviewer 2: I think it would be helpful if we followed the [Newman documentation](https://learning.postman.com/docs/running-collections/using-newman-cli/newman-options/) format/terms for users wanting to switch over and/or compare the two. For example, in Newman we use "Details" instead of "Description".

I'm wondering if we should separate the commands and options into their own sections. I like what we did with the Newman options where we separate sections by option type. I'm mentioning this because I was not clear why we have sections titled "postman" and "postman logout". Giving these sections a parent header like "Commands" could make this clearer.

|:--|:--|

> | [**`postman`**](#postman) | The base command, commonly used with other commands like `login` or `collection`. |

Reviewer 2: Is `postman` itself a command? Or would a command refer more to something like `logout` that is entered as `postman logout`?

> | `--help` | Return information about Postman CLI commands and options. |

Reviewer 2: Can users also enter `-h`?

Author: Yes but I deliberately left out the short commands in the table at the top for brevity. I plan to update the the more verbose descriptions with long and short commands, but that'll happen later when the commands and options stop churning so much.

> | `--version` | Return the version number for the Postman CLI.  |

Reviewer 2: Can users also enter `-v`?

| [**`postman login`**](#postman-login---with-api-key-api-key) | Log into Postman. |
| [**`postman logout`**](#postman-logout) | Log out of Postman. |

> | [**`postman collection run`**](#postman-collection-run-uuid-or-file-name) | Run a collection with a UUID or a local file. |

Reviewer 2: The relative URL does not seem correct.

| `--bail` | Stop the runner when a test case fails. |

> | `--color` | Turn off colored output (auto\|on\|off) (default: "auto") |

Reviewer 2: How would this be entered on the command line? Would an example be `postman --color off`?
If so, I'm wondering if "auto", "on", and "off" should be formatted as inline code samples.

| `--cookie-jar` | Specify the file path for a JSON Cookie Jar. Uses `tough-cookie` to deserialize the file. |

> | `--delay-request` | Specify a delay (in milliseconds) between requests [number]. |

Reviewer 2: Why is "number" between brackets?

Reviewer 2: Looking at the Newman docs, it looks like the command should be written as `--delay-request [number]`.

Author: Typo, fixed.

| `--disable-unicode` | Force the unicode disable option. When supplied, replaces all symbols in the output with their plain text equivalents. |
| `--env-var` | Set environment variables in a key=value format on the command line. |

> | `--environment` | Specify a Postman environment as a JSON [file]. |

Reviewer 2: Why is "file" between brackets?

Reviewer 2: Looking at the Newman docs, it looks like the command should be written as `--environment [file]`.

| `--silent` | Turn off terminal output. |
| `--suppress-exit-code` | Continue running tests even after a failure, but exit with `code=0` |
| `--timeout` | Specify the time (in milliseconds) to wait for the entire collection run to complete execution. |
| `--timeout-request` | Specify a request timeout (in milliseconds) for a request. |
| `--timeout-script` | Specify the time (in milliseconds) to wait for scripts to complete execution. |
| `--verbose` | Show the details of the collection run and its data ingestion to Postman. |
| `--working-dir` |Set the path of the working directory to use while reading files with relative paths. |
| `-r` | Run a collection with a custom reporter. |
| [**`postman api lint`**](#postman-api-lint-uuid-or-file-name)| Run validation checks for governance and security rules against the api definition provided in the Postman config file. |
| `--disable-postman-reporter` | Don't upload data to Postman after linting. |
| `--only-governance-rules` | Only use governance rules at the time of linting. |
| `--only-security-rules` | Only use security rules at the time of linting. |

> ## Downloading the installation package

Reviewer 2: I believe we touch upon these instructions at `/postman-cli/postman-cli-installation/`. Should we keep download/install instructions on the same page?

| `--timeout-request` | Specify a request timeout (in milliseconds) for a request. |
| `--timeout-script` | Specify the time (in milliseconds) to wait for scripts to complete execution. |
| `--verbose` | Show the details of the collection run and its data ingestion to Postman. |
| `--working-dir` |Set the path of the working directory to use while reading files with relative paths. |
| [**`postman api lint`**](#postman-api-lint)| Run validation checks for governance and security rules against the api definition provided in the Postman config file. |
| [**`postman`**](#postman) | The base command. |
| `--help` | Return information about Postman CLI commands and options. |
| `--version` | Return the version number for the Postman CLI.  |

## Downloading and installing

You can download the Postman CLI installation script manually or programmatically (from the command line or a script).

### Downloading manually

To download manually, use the commands and links below:

#### Linux

`curl -o- "https://[CDN link TBD]/install.linux64.sh" | bash`

#### Mac (Intel chip)

`curl -o- "https://[CDN link TBD]/install.osx_64.sh" | bash`

#### Mac (Apple chip)

> `curl -o- "https://[CDN link TBD]/install.osx_arm64.sh" | bash`

Reviewer 2: Should the code be formatted as code blocks, not inline code samples?

Author: Now formatted as blocks.

postman collection run 12345678-12345ab-1234-1ab2-1ab2-ab1234112a12
```

#### Options

| Option | Details |
|:--|:--|
| `--bail [optional modifiers]` | Specifes whether or not to stop a collection run on encountering the first test script error. `--bail` can optionally accept two modifiers: `--folder` and `--failure`. `--folder` skips the entire collection run if there are any errors. If a test fails, `--failure` gracefully stops the collection run after completing the current test script. |
| `--color [value]` | Controls colored CLI output. Accepts `on`, `off`, and `auto`. Default is `auto`. With `auto`, Postman CLI attempts to automatically turn color on or off based on the color support in the terminal. This behavior can be modified by using the on or off value accordingly.|
| `--cookie-jar [path]` | Specifies the file path for a `JSON` cookie jar. Uses `tough-cookie` to deserialize the file. |
| `--delay-request [number]` | Specifies a delay (in milliseconds) between requests. |
| `--disable-unicode` | Replaces all symbols in the output with their plain text equivalents. |
| `--environment [uid] or [file-path]`, `-e` | Specifies an environment file path or UID. |

> | `--env-var "[environment-variable-name]=[environment-variable-value]"` | Specifies environment variables via the command line, in a `key=value` format. Multiple CLI environment variables can be added by using `--env-var` multiple times, for example: `--env-var "this=that" --env-var "alpha=beta".`|

Reviewer 2: ```suggestion
| `--env-var "[environment-variable-name]=[environment-variable-value]"` | Specifies environment variables via the command line, in a `key=value` format. Multiple CLI environment variables can be added by using `--env-var` multiple times, for example: `--env-var "this=that" --env-var "alpha=beta"`.|
```

> | `--env-var "[environment-variable-name]=[environment-variable-value]"` | Specifies environment variables via the command line, in a `key=value` format. Multiple CLI environment variables can be added by using `--env-var` multiple times, for example: `--env-var "this=that" --env-var "alpha=beta".`|

Author: Fixed

| `--ignore-redirects` | Prevents the Postman CLI from automatically following 3XX redirect responses.|
| `--insecure`, `-k` | Turns off SSL verification checks and allows self-signed SSL certificates. |
| `--no-insecure-file-read` | Prevents reading of files situated outside of the working directory.|
| `--silent` | Turns off terminal output.|
| `--suppress-exit-code`, `-x`| Specify whether or not to override the default exit code for the current run.|
| `--timeout [number]`| Specifies the time (in milliseconds) to wait for the entire collection run to complete execution.|
| `--timeout-request [number]`| Specifies a time (in milliseconds) to wait for requests to return a response.|
| `--timeout-script [number]`| Specifies the time (in milliseconds) to wait for scripts to complete execution.|
| `--verbose` | Shows detailed information of collection run and each request sent.|
| `--working-dir [path]`| Sets the path of the working directory to use while reading files with relative paths. Defaults to current directory.

---

## Governance and security

> API governance is the practice of applying a defined set of standards consistently across your API design and testing phases of your development process. The Postman CLI includes an api linting command, described below.

Reviewer 2: ```suggestion
API governance is the practice of applying a defined set of standards consistently across your API design and testing phases of your development process. The Postman CLI includes an API linting command, described below.
```

> ### postman api lint

Reviewer 2: In the "postman apit lint" section, we include a list of options. I would expect to also see an "Options" section here that includes `api lint`. What do you think?

Author: api lint doesn't have any options, it just accepts a UUID or a filepath.

> This command runs validation checks for governance and security rules against the api definition provided in the Postman config file, a local file, or a UUID. You can only lint single-file definitions. `api lint` shows a warning if unable to find `[api-id]` to send data back to Postman.

Reviewer 2: Instead of `[api-id]`, what is we write "the API ID"? The relationship between `api lint` and `[api-id]` is not clear to me.

Author: Fixed

> This command supports APIs that are stored on Postman and aren't linked to Git.

#### Example

```
postman api lint my-definition-file.json
postman api lint 8854915-bb7236b2-536e-4bdc-bfa2-fbe2fe1941eb
```

---

## Basic command line options

> ### postman

Reviewer 2: Should this section be moved towards the top of the page? I would expect to learn about the basic command before the more complex commands.
